File: lambda/go/experiment1/matrixMultiplication.go
package main

import (
	"context"

	"github.com/aws/aws-lambda-go/lambda"

	"math/rand/v2"
)

func main() {
	lambda.Start(placeholder)
}

type Response struct {
	StatusCode int `json:"statusCode"`
}

func placeholder(ctx context.Context) (Response, error) {
	//generate random matrix dimensions
	A_rows := rand.IntN(11) + 90
	A_cols := rand.IntN(11) + 90
	B_rows := A_cols
	B_cols := rand.IntN(11) + 100

	//generate random matrices
	A := make([][]int, A_rows)
	for i := range A {
		A[i] = make([]int, A_cols)
		for j := range A[i] {
			A[i][j] = rand.IntN(128)
		}
	}

	B := make([][]int, B_rows)
	for i := range B {
		B[i] = make([]int, B_cols)
		for j := range B[i] {
			B[i][j] = rand.IntN(128)
		}
	}

	//allocate result matrix
	result := make([][]int, A_rows)
	for i := range result {
		result[i] = make([]int, B_cols)
	}

	//perform multiplication
	for x := range A_rows {
		for y := range B_cols {
			for z := range B_rows {
				result[x][y] += A[x][z] * B[z][y]
			}
		}
	}

	return Response{
		StatusCode: 200,
	}, nil
}

File: lambda/python/experiment1/matrixMultiplication.py
import random

def multiply_matrix():
    #generate random matrix dimensions
    A_rows = random.randint(90,100)
    A_cols = random.randint(90,100)
    B_rows = A_cols
    B_cols = random.randint(100,110)
    
    #generate random matrices
    A = [[random.randint(0,127) for _ in range(A_cols)] for _ in range(A_rows)]
    B = [[random.randint(0,127) for _ in range(B_cols)] for _ in range(B_rows)]

    #allocate result matrix
    result = [[0] * len(B[0]) for _ in range(len(A))]

    #perform multiplication
    for x in range(len(A)):
        for y in range(len(B[0])):
            for z in range(len(B)):
                result[x][y] += A[x][z] * B[z][y]

    return result